holoviews/core/data/__init__.py

```diff
@@ -188,19 +193,29 @@ def __setstate__(self, state):
 
         super(Dataset, self).__setstate__(state)
 
-    def closest(self, coords):
+
+    def closest(self, coords=[], **kwargs):
         """
         Given single or multiple samples along the first key dimension
         will return the closest actual sample coordinates.
         """
         if self.ndims > 1:
-            NotImplementedError("Closest method currently only "
-                                "implemented for 1D Elements")
+            raise NotImplementedError("Closest method currently only "
+                                      "implemented for 1D Elements")
+
+        if kwargs:
+            dim = self.get_dimension(list(kwargs.keys())[0], strict=True)
+            if len(kwargs) > 1:
+                raise NotImplementedError("Closest method currently only "
+                                          "supports 1D indexes")
+            samples = list(kwargs.values())[0]
+            coords = samples if isinstance(samples, list) else [samples]
 
-        if not isinstance(coords, list): coords = [coords]
         xs = self.dimension_values(0)
+        if xs.dtype.kind in 'SO':
+            raise NotImplementedError("Closest only supported for numeric types")
         idxs = [np.argmin(np.abs(xs-coord)) for coord in coords]
-        return [xs[idx] for idx in idxs] if len(coords) > 1 else xs[idxs[0]]
+        return [xs[idx] for idx in idxs]
```

Comment:
Looks like the return format has changed given only a single sample. The new version does seem more consistent.